src/index.ts

```diff
@@ -203,19 +154,21 @@ function setCumulusEnvironment(cumulusMessage, context) {
  * @returns {Promise<Object>} - Promise that resolves to a parsed JSON object
  *                              from the CMA output
  */
-async function getCmaOutput(readLine, errorObj) {
+async function getCmaOutput(
+  readLine: readline.ReadLine,
+  errorObj: cumulusMessageAdapterError
+): Promise<cumulusMessageWithGranulesInPayload | loadNestedEventInput | CumulusRemoteMessage> {
```

Comment:
Given that this function doesn't actually know what it's returning, but the calling code does, it might make more sense for this to return `unknown` and have the calling code handle casting.

Reply:
🤔   I suppose you're right.    We're explicitly using a runtime typeguard here, so that works.   

I was erring on the side of explicit over implicit here as the return type is defined by an external utility/contract that *cannot* be typed directly, but..... I suppose that's being redundant.

Reply:
On second thought, I'm not sure that's true.   Instead of filtering between options in a union type (what the CMA call is really returning by convention/contract) , modifying it in that way means we are instead making the typeguard take *any* object.  That feels like a possible anti-pattern as while it certainly works, it allows a future developer to isLoadNestedEvent(<some other thing that isn't even remotely intended to be checked>). 

https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types


I'm sure I'm missing an approach, or misunderstanding something -  perhaps we should chat re: this.

Reply:
I should learn about type guards. Standby.